scripts/eqtl_hail_batch/conditional_analysis.py

```diff
@@ -77,17 +77,20 @@ def get_genotype_df(residual_df, gene_snp_test_df):
     # only keep samples that are contained within the residuals df
     # this is important, since not all indivuduals have expression/residual
     # data (this varies by cell type)
-    samples_to_keep = set(residual_df.sampleid)
-    set_to_keep = hl.literal(samples_to_keep)
-    mt = mt.filter_cols(set_to_keep.contains(mt['onek1k_id']))
+    samples_to_keep = hl.literal(list(residual_df.sampleid))
```

Comment:
`contains` is likely faster on a `set` compared to a list. Not sure if you need to first convert to a list (`hl.set(list(residual_df.sampleid))`) though.

```suggestion
    samples_to_keep = hl.set(residual_df.sampleid)
```

Reply:
hl.set(list(residual_df.sampleid)) works 👌